lib/tofuenv-version-name.sh

```diff
@@ -10,10 +10,14 @@ function tofuenv-version-name() {
       && log 'debug' "TOFUENV_VERSION_FILE retrieved from tofuenv-version-file: ${TOFUENV_VERSION_FILE}" \
       || log 'error' 'Failed to retrieve TOFUENV_VERSION_FILE from tofuenv-version-file';
 
-    TOFUENV_VERSION="$(cat "${TOFUENV_VERSION_FILE}" || true)" \
-      && log 'debug' "TOFUENV_VERSION specified in TOFUENV_VERSION_FILE: ${TOFUENV_VERSION}";
-
-    TOFUENV_VERSION_SOURCE="${TOFUENV_VERSION_FILE}";
+    if [[ -f "${TOFUENV_VERSION_FILE}" ]]; then
+        TOFUENV_VERSION="$(cat "${TOFUENV_VERSION_FILE}" || true)" \
+            && log 'debug' "TOFUENV_VERSION specified in TOFUENV_VERSION_FILE: ${TOFUENV_VERSION}";
+        TOFUENV_VERSION_SOURCE="${TOFUENV_VERSION_FILE}";
+    else
+        TOFUENV_VERSION="${TOFUENV_TOFU_DEFAULT_VERSION:-"latest"}";
```

Comment:
Continuing my comment in the discussion, I suggest setting the version to an empty string by default, ex.
`TOFUENV_VERSION="${TOFUENV_TOFU_DEFAULT_VERSION:-""}";`

Reply:
@anastasiiakozlova245 Won't that mean that `tofuenv exec` will not have the correct/consistent behaviour?

Reply:
@orbitz could you please clarify what you mean by not having a consistent behaviour? As stated in the docs, `tofuenv install` command will still install the latest version if no version is specified, since it doesn't use this particular script. However, `tofuenv version-name` should show the version of `tofu` currently installed/set in configs (`.opentofu-version` file, env variables). If `tofuenv` is just installed, and no version is set anywhere, why should it print the latest one?

Reply:
Perhaps I'm confused bu but `tofuenv-exec` which is called from the `tofu` wrapper does use this function.  I would expect it to have the semantics as `tofuenv install`.  If I change line 18 to `:-""` then if no version is found it will look for version "", which will never exist.  For example, in `main`, if I have no version information set:

```
> ./bin/tofu version
cat: /usr/home/orbitz/projects/terrateam/tofuenv/version: No such file or directory
Version could not be resolved (set by /usr/home/orbitz/projects/terrateam/tofuenv/version or tofuenv use <version>)
```

However, with this change:

```
> ./bin/tofu version
version '1.6.1' is not installed (set by /usr/home/orbitz/projects/terrateam/tofuenv/version). Installing now as TOFUENV_AUTO_INSTALL==true
Installing OpenTofu v1.6.1
Downloading release tarball from https://github.com/opentofu/opentofu/releases/download/v1.6.1/tofu_1.6.1_freebsd_amd64.zip
##################################################################################################################################################################################################### 100.0%
Downloading SHA hash file from https://github.com/opentofu/opentofu/releases/download/v1.6.1/tofu_1.6.1_SHA256SUMS
Not instructed to use Local GPG (/usr/home/orbitz/projects/terrateam/tofuenv/use-{gpgv,gnupg}), skipping GnuPG signature verification

Installation of tofu v1.6.1 successful. To make this your default version, run 'tofuenv use 1.6.1'
OpenTofu v1.6.1
on freebsd_amd64
```

This is the behaviour I would expect, is this not the behaviour you would expect?  If so, what is?

Reply:
@orbitz I believe when you call `./bin/tofu version`, it shouldn't trigger an automatic installation of a latest package. `version` command is not supposed to install anything, it should just provide an information about the version installed or set anywhere in configs. And if it's not - they returning an error is expected.
@Nmishin what do you think the correct behaviour would be?

Reply:
I think there are a few things to consider:

1. Even if you don't think an install should happen, I believe that calling `tofu foo` should get the same version as when you called `tofuenv install`.  So if `tofuenv install` defaults to `latest` if no version information is set, then `tofu foo` should get that value as well.
2. As it stands now, if you DO set any version information then `tofu foo` will install that version if it is not set.  Line 18 just guarantees that it is matches the `tofuenv install` version discovery algorithm.
3. As the code stands now, whether or not `tofu foo` installs something depends on the  value of the auto install variable.

Reply:
Just bumping this discussion.